test/internal_benchmarks/eof_jumpdest_section_bench.cpp

```diff
@@ -0,0 +1,99 @@
+
+#include <benchmark/benchmark.h>
+#include <cstdint>
+#include <limits>
+#include <memory>
+#include <utility>
+
+template <typename T>
+inline std::pair<T, const uint8_t*> leb128u_decode_safe(const uint8_t* pos, const uint8_t* end)
+{
+    static_assert(!std::numeric_limits<T>::is_signed);
+
+    T result = 0;
+    int result_shift = 0;
+
+    for (; result_shift < std::numeric_limits<T>::digits; ++pos, result_shift += 7)
+    {
+        if (pos == end)
+            throw "unexpected EOF";
+
+        result |= static_cast<T>((static_cast<T>(*pos) & 0x7F) << result_shift);
+        if ((*pos & 0x80) == 0)
+        {
+            if (*pos != (result >> result_shift))
+                throw "invalid LEB128 encoding: unused bits set";
+
+            return {result, pos + 1};
+        }
+    }
+
+    throw "invalid LEB128 encoding: too many bytes";
+}
+
+template <typename T>
+inline std::pair<T, const uint8_t*> leb128u_decode_fast(const uint8_t* pos) noexcept
+{
+    static_assert(!std::numeric_limits<T>::is_signed);
+
+    T result = 0;
+    int result_shift = 0;
+
+    for (; result_shift < std::numeric_limits<T>::digits; ++pos, result_shift += 7)
+    {
+        result |= static_cast<T>((static_cast<T>(*pos) & 0x7F) << result_shift);
+        if ((*pos & 0x80) == 0)
+            break;
+    }
+
+    return {result, pos + 1};
+}
+
+static std::vector<bool> load_jumpdests_safe(
```

Comment:
Can't you templetize this since the only difference is using `leb128u_decode_safe` vs` leb128u_decode_fast`?

Reply:
Can be, though they use different number of arguments.